src/stravalib/protocol.py

```diff
@@ -89,6 +102,168 @@ def __init__(
             lambda _request_params, _method: None
         )
 
+        # mypy is flagging this method. Not sure why.
+
+    def _request(
+        self,
+        url: str,
+        params: dict[str, Any] | None = None,
+        files: dict[str, SupportsRead[str | bytes]] | None = None,
+        method: RequestMethod = "GET",
+        check_for_errors: bool = True,
+        refresh_expired_token: bool = True,
+    ) -> Any:
+        """Perform the underlying request, returning the parsed JSON results.
+
+        This method before running will check to make sure that your
+        access_token is valid. If it isn't, it will refresh it and then make
+        the requested Strava API call.
+
+        Parameters
+        ----------
+        url : str
+            The request URL.
+        params : Dict[str,Any]
+            Request parameters
+        files : Dict[str,file]
+            Dictionary of file name to file-like objects.
+        method : str
+            The request method (GET/POST/etc.)
+        check_for_errors : bool
+            Whether to raise an error or not.
+        refresh_expired_token : bool (default True)
+            Whether to automagically refresh the users token or not.
+            Requires environment setup to work.
+
+        Returns
+        -------
+        Dict[str, Any]
+            The parsed JSON response.
+        """
+        # Try to refresh the token unless told to not try
+        if refresh_expired_token:
```

Comment:
To prevent that infinite loop, could we check the URL at this point and ignore refreshing if the URL itself ends with `/oauth/token`? In that case, we could leave the existing `exchange_code_for_token()` method unchanged.

Reply:
Ahhhh so much better. One less parameter as well in this method. Thanks!!